src/main/scala/io/github/sentenza/hacktoberfest18/algos/MutableSorting.scala

```diff
@@ -185,10 +188,41 @@ package io.github.sentenza.hacktoberfest18.algos
       array
     }
 
-    def bucketSort(xs: Array[Int]): Array[Int] = ???
+  /**
+    * @author Xoeseko
+    * Bucket sort is a sorting algorithm that sorts an array of elements by splitting the elements into n buckets
+    * and then reapplies another sorting method on the different buckets. Then merges the sorted buckets.
+    * It can also be used recursively until buckets contain one element each.
+    *
+    * @param xs Array of sortable integers
+    * @param n number of buckets in which to sort the elements.
+    * @param sort The sorting algorithm to apply once buckets are divided.
+    * @return the sorted array
+    */
+  def bucketSort(xs: Array[Int], n: Int = 10, sort: Array[Int] => Array[Int] = insertionSort): Array[Int] = {
+    val buckets = new Array[Array[Int]](n)
+    val temp: ArrayBuffer[Int] = new ArrayBuffer[Int]()
+    val range: Int = (xs.max / n)+1
+    temp ++= xs
+
+    if(temp.length > 1) {
+      for (i <- 0 until n) {
+        buckets(i) = sort(temp.filter(a => a >= range*i && a < range*(i+1)).toArray)
+      }
+    } else return xs
+
+    var finalArray: ArrayBuffer[Int] = new ArrayBuffer[Int]()
```

Comment:
once more
```scala
val finalArray = ArrayBuffer[Int]()
```

Reply:
Same as previous I am curious why you think that typing is a bad idea ?